src/utils/keys.ts

```diff
@@ -1,73 +1,54 @@
-import bsv from 'bsv';
-import Mnemonic from 'bsv/mnemonic';
 import { getChildNumsFromHex } from './index';
-
 import { Key, KeyWithMnemonic } from '../types';
+import { HD, Mnemonic } from '@bsv/sdk';
 
 // deriveChildKeyFromHex derive the child extended key from the hex string
 export const deriveChildKeyFromHex = function (
-  hdKey: bsv.HDPrivateKey | bsv.HDPublicKey,
+  hdKey: HD,
   hexHash: string,
-): bsv.HDPrivateKey | bsv.HDPublicKey {
-  if (hdKey instanceof bsv.HDPrivateKey) {
-    return deriveHDPrivateChildKeyFromHex(hdKey as bsv.HDPrivateKey, hexHash);
-  } else {
-    return deriveHDPublicChildKeyFromHex(hdKey, hexHash);
-  }
-};
-
-export const deriveHDPrivateChildKeyFromHex = function (hdKey: bsv.HDPrivateKey, hexHash: string): bsv.HDPrivateKey {
-  let childKey: bsv.HDPrivateKey = hdKey;
-  const childNums = getChildNumsFromHex(hexHash);
-
-  childNums.forEach((childNum) => {
-    childKey = childKey.deriveChild(childNum, false);
-  });
-
-  return childKey;
+): HD{
+  return deriveHDChildKeyFromHex(hdKey, hexHash);
 };
 
-export const deriveHDPublicChildKeyFromHex = function (hdKey: bsv.HDPublicKey, hexHash: string): bsv.HDPublicKey {
-  let childKey: bsv.HDPublicKey = hdKey;
+export const deriveHDChildKeyFromHex = function (hdKey: HD, hexHash: string): HD {
+  let childKey: HD = hdKey;
   const childNums = getChildNumsFromHex(hexHash);
-
   childNums.forEach((childNum) => {
-    childKey = childKey.deriveChild(childNum, false);
+    childKey = childKey.deriveChild(childNum);
   });
 
   return childKey;
 };
 
 export const generateKeys = function (): KeyWithMnemonic {
-  bsv.HDPrivateKey.fromRandom();
   const mnemonic = Mnemonic.fromRandom();
   return getKeysFromMnemonic(mnemonic.toString());
 };
 
 export const getKeysFromMnemonic = function (mnemonicStr: string): KeyWithMnemonic {
   const mnemonic = Mnemonic.fromString(mnemonicStr);
   const seed = mnemonic.toSeed();
-  const hdPrivateKey = bsv.HDPrivateKey.fromSeed(seed, bsv.Networks.mainnet);
```

Comment:
Unfortunately, it looks like those Mnemonic methods are not working properly.
Ex.: 
`const mnemonic = "they veteran need deliver refuse embrace isolate leaf profit regular project you asset taxi describe";`

Produces this:
`63189359842774274555045717767992799560554611297080512701541582869260668002353 `

instead of this:
`xprv9s21ZrQH143K2LpsdRco6N9RvxvxwaWoBK9i4dk2hVwJ9tBPr1qQY6Bxo7dfK2QpevbCFaCFGekAoSPxjLxwvYyUhHXkL5RUct2m7dpf3ZX `.

Our issue to track this problem: **SPV-766**